webots_ros2_epuck/README.md

```diff
@@ -129,16 +129,74 @@ Ground sensors come as an [optional module](http://www.e-puck.org/index.php?opti
 ``` 
 ros2 topic echo /gs1
 ```
-To put the ground sensor module, select `groundSensorsSlot` in `e-puck2` robot tree, click `+` button and find `E-puckGroundSensors` (check the image bellow).
-![e-puck2 ground sensors](./assets/ground_sensors_webots.png)
+To put the ground sensor module, select `groundSensorsSlot` in `e-puck2` robot tree, click `+` button and find `E-puckGroundSensors` (check out [this image](./assets/ground_sensors_webots.png)).
 
 ### Transformations
 Dynamic transformations are only used for the odometry and you can show it as:
 ```
 ros2 topic echo tf
 ```
 
-All other transformations are static and they are exposed as latched topics, so you show them with the following command:
+All other transformations are static and they are exposed as latched topics, so you can show them with the following command:
 ```
 ros2 topic echo --qos-profile services_default --qos-durability transient_local tf_static
+```
+
+For general access to transformations you can use `tf2_monitor`:
+```
+ros2 run tf2_ros tf2_monitor
+```
+or if you want to read transformation between arbitrary two coordinate frames in a tree:
+```
+ros2 run tf2_ros tf2_echo odom map
+```
+
+
+### Navigation
+ROS2 Navigation2 stack (see [this figure](https://raw.githubusercontent.com/ros-planning/navigation2/eloquent-devel/doc/architecture/navigation_overview.png)) allows us to move robot from point A to point B by creating a global plan and avoiding local obstacles. It is integrated into e-puck example and you can run it by including `nav` parameter:
+
+```
+ros2 launch webots_ros2_epuck example_launch.py rviz:=true nav:=true mapper:=true fill_map:=false
+```
+or without RViz2 you can just publish a desired pose:
+```
+ros2 topic pub -1 /goal_pose geometry_msgs/PoseStamped \"
+pose:
+  position:
+    x: 0.3
+    y: 0.0
+    z: 0.0
+  orientation:
+    x: 0.0
+    y: 0.0
+    z: 0.0
+    w: 1.0
+"
+```
+
+![Demo](./assets/nav2.gif)
+
+This example will work properly only for [ROS2 Foxy](https://index.ros.org/doc/ros2/Releases/Release-Foxy-Fitzroy/) (the first ROS2 release that has a long support - 3+ years):
+- Navigation2 stack is tested with version `e3469486675beb3` that includes [fix of progress checker parameters](https://answers.ros.org/question/344004/configuring-the-progress-checker-in-navigation2/) and [namespaced plugins for servers](https://github.com/ros-planning/navigation2/pull/1468).
+- RViz2 for Eloquent has a bug and [cannot show a local cost map](https://github.com/ros-planning/navigation2/issues/921), therefore, `ff8fcf9a2411` (or up) version of RViz2 is desired.
```

Comment:
Is there any possibility to test this PR without having to recompile any package manually?

Reply:
> I am wondering about the `example_tools_launch` and `simple_mapper, they have nothing specific to the e-puck right?
> In that case, maybe we can think about putting them in a more generic package, what do you think?

Hmm, maybe, RViz2 and Navigation2 configuration files (Navigation2 configuration file is really tuned for e-puck) are made for e-puck. Therefore, those two files should be parameters. `simple_mapper` doesn't make sense for robots that have proper sensors, but that is already optional.

In short, with RViz2 and Navigation2 configurable (config filepath) it would work.

Reply:
Ok in that case I will recompile them.

Reply:
BTW, I am sketching ROS2 package manager that should be able to install packages from sources more easily:
https://gist.github.com/lukicdarkoo/d03196b15367b804b27c142dad8644e9

It may be helpful in similar situations